parser_library/test/context/literals_test.cpp

```diff
@@ -289,9 +277,8 @@ TEST(literals, processing_stack_in_messages)
 )";
     analyzer a(input);
     a.analyze();
-    a.collect_diags();
 
-    auto& d = a.diags();
+    auto d = a.diags();
     ASSERT_TRUE(matches_message_codes(d, { "E010" }));
```

Comment:
Can we use the following here as well?

```suggestion
    ASSERT_TRUE(matches_message_codes(a.diags(), { "E010" }));
```